cpp/src/text/subword/wordpiece_tokenizer.cu

```diff
@@ -82,7 +82,9 @@ __global__ void init_data_and_mark_word_start_and_ends(uint32_t const* code_poin
                                                        uint32_t* token_ids,
                                                        uint8_t* tokens_per_word)
 {
-  uint32_t char_for_thread = blockDim.x * blockIdx.x + threadIdx.x;
+  cudf::thread_index_type char_for_thread = static_cast<cudf::thread_index_type>(blockDim.x) *
```

Comment:
Minor inconsistency in this PR about whether to use `static_cast<cudf::thread_index_type>(x)` or `cudf::thread_index_type(x)`. It could also be `cudf::thread_index_type{x}`. I don't have a preference, but it might be nice to use the same style.